backup/manager.go

```diff
@@ -241,13 +249,66 @@ func (b *Manager) IsSafeToRunNode(nodeID string) (bool, error) {
 	if err != nil {
 		return false, err
 	}
+
 	for _, s := range snapshots {
 		if s.NodeID == nodeID && s.BackupID != "" && backupID != s.BackupID {
 			b.log.Errorf("remote restore was found for node %v.", nodeID)
 			b.log.Errorf("current backupID=%v, remote backupID-%v", backupID, s.BackupID)
 			return false, nil
 		}
 	}
+
+	return true, nil
+}
+
+// ensureLatestBackup checks whenever a given local state is newer than our snapshot.
+// if there is no local state, it returns true.
+func (b *Manager) EnsureLatestBackup(nodeID string) (bool, error) {
+
+	provider := b.GetProvider()
+	if provider == nil {
+		return true, ErrorNoProvider
+	}
+	snapshots, err := provider.AvailableSnapshots()
+	if err != nil {
+		return true, err
+	}
+
+	providerName, err := b.getProviderName()
+	if err != nil {
+		return true, err
+	}
+	b.log.Infof("Using provider: %v", providerName)
+
+	var snapshot SnapshotInfo
+	for _, s := range snapshots {
+		if s.NodeID == nodeID {
+			snapshot = s
+		}
+	}
+
+	if snapshot.ModifiedTime.IsZero() {
+		return true, nil
+	}
+
+	latestBackup, err := b.db.latestBackupTime(providerName)
+	if err != nil {
+		b.log.Errorf("error getting latestBackup exited with err: %v", err)
+		if !strings.Contains(err.Error(), "no backup time found") {
+			return true, err
```

Comment:
If there is no backup time this mean we didn't backup using this node but there is a backup in the cloud. Isn't it a problematic case?
I am thinking a user started a new app, iOS has backed up, then the user backed up using breez.
After that the user has restored the old snapshot with the iOS system resulting in a stale state.

Reply:
Agree this can be a problematic case, the reason for me adding it was due to these processes `b.updateLatestBackupTime(nodeID)` and `ensureLatestBackup` are running asynchronously so a race condition may emerge.